src/test/scala/parsing/extractHeadersTests.scala

```diff
@@ -0,0 +1,153 @@
+    val parseInputSource2: ParserInput = ParserInput(
+      in = IteratorType(source2),
+      csvDefinition = formats.PSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val getHeader: ParserInput = extractHeaders(parseInputSource)
+    getHeader.headers shouldBe Iterator("Ballon Dor List").toList
+
+    val getHeaderSource1: ParserInput = extractHeaders(parseInputSource1)
+    getHeaderSource1.headers shouldBe Iterator("Ballon Dor List").toList
+
+    val getHeaderSource2: ParserInput = extractHeaders(parseInputSource2)
+    getHeaderSource2.headers shouldBe Iterator("\"Ballon Dor List\"|\"club\"\n\"Karim Benzema\"|\"Real Madrid\"\n\"Sadio Mane\" | \"Liverpool\"\n\"Kevin De Bruyne\" | \"Manchester  City\"\n\"Robert Lewandowski\" | \"Barcelona\"\n\"Mohamed Salah\" | \"Liverpool\"\n\"Ricardo Esgaio\" | \"Sporting\"\n\"Mbappe\" | \"PSG\"").toList
+  }
+
+  "Seq" should "return header" in {
+
+    val source: Seq[String] = Seq("Ballon dor","Karim Benzema")
+    val source1: Seq[String] = Seq("Ballon dor,,,,,Ballon dor feminin,Kopa Trophy,,,,Yashing Trophy,","Karim Benzema, Alexia Putellas, Gavi, Courtois")
+
+    val parseInputSource: ParserInput = ParserInput(
+      in = SeqType(source),
+      csvDefinition = formats.CSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val parseInputSource1: ParserInput = ParserInput(
+      in = SeqType(source1),
+      csvDefinition = formats.CSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val getHeader: ParserInput = extractHeaders(parseInputSource)
+    getHeader.headers shouldBe Seq("Ballon dor").toList
+
+    val getHeader1: ParserInput = extractHeaders(parseInputSource1)
+    getHeader1.headers shouldBe Seq("Ballon dor,,,,,Ballon dor feminin,Kopa Trophy,,,,Yashing Trophy,").toList
+  }
+
+  "Seq with CSV format from CSVdefinition " should "return the headers" in {
+
+    val source: Seq[String] = Seq("Ballon dor\n\n\n\n\n,,,\\\\\\\\,,Ballon dor feminin,Kopa Trophy,\r\n,,\r\n,Yashing Trophy,","Karim Benzema, Alexia Putellas, Gavi, Courtois")
+    val parseInputSource: ParserInput = ParserInput(
+      in = SeqType(source),
+      csvDefinition = formats.CSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val source1: Seq[String] = Seq("Ballon dor\n\n\n\n\n,,,\\\\\\\\,,Ballon dor feminin,Kopa Trophy,\r\n,,\r\n,Yashing Trophy,,,,\n\n\n\n\r\n\r\n","Karim Benzema, Alexia Putellas, Gavi, Courtois")
+    val parseInputSource1: ParserInput = ParserInput(
+      in = SeqType(source1),
+      csvDefinition = formats.CSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val getHeader: ParserInput = extractHeaders(parseInputSource)
+    getHeader.headers shouldBe Seq("Ballon dor\n\n\n\n\n,,,\\\\\\\\,,Ballon dor feminin,Kopa Trophy,\r\n,,\r\n,Yashing Trophy,").toList
```

Comment:
Just a small thing: You can create a `List(...)` directly instead of `Seq(...).toList`

Reply:
thank you, done